### DISCUSSION TITLE
How to Choose an Optimization Function

### DISCUSSION THREAD

**Discussion by kkmcgg** [NSCC-COGS/Aestheta#66] [Guides]
https://keras.io/api/optimizers/

Optimizers are the hard working part of the neural net. They are responsible for navigating the loss landscape such that we can iteratively nudge our model down the loss gradient and improve the model per epoch toward some nice local minimum of loss. This navigation is what we generally call **_gradient descent_**. Without a good optimizer, our models wouldn't get better! They would wander around adjusting random weights and never converge on a reasonable solution. Conversely a bad optimizer may work _much_ too hard and take a _very_ long time to converge. So it is a trade off between effective and efficient navigation. This guide shows that there are several options for optimizers to choose from in keras. 

The basic TLDR of this guide is to **_choose adam_** (`optimizer='adam'`) which generally has a good mix of adaptable learning rates and smooth gradient descent. Other good options may include **_stochastic gradient descent (SGD)_**

**Comment by kkmcgg** [comment]
Some additional resources:

On optimizers in general: https://mlfromscratch.com/optimizers-explained/#/
Details on the adam optimizer: https://machinelearningmastery.com/adam-optimization-algorithm-for-deep-learning/
